compiler/src/dmd/toobj.d

```diff
@@ -785,6 +785,53 @@ void toObjFile(Dsymbol ds, bool multiobj)
                     global.params.libfiles.push(name);
                 }
             }
+            else if (pd.ident == Id.libLocal)
+            {
+                import dmd.root.filename: FileName;
+                assert(pd.args && pd.args.length == 1);
+
+                Expression e = (*pd.args)[0];
+
+                assert(e.op == EXP.string_);
+
+                StringExp se = cast(StringExp)e;
+                char *name = cast(char *)mem.xmalloc(se.numberOfCodeUnits() + 1);
```

Comment:
The numberOfCodeUnits for a char string is simply its length. Even better, just use StringExp.peekString() and avoid the allocation for `name`.